include/lsst/afw/geom/SkyWcs.h

```diff
@@ -94,158 +100,391 @@ Constructors must set the following properties of the contained ast::FrameSet:
+ *
+ * If modifyActualPixels is true and the ACTUAL_PIXELS frame exists then pixelTransform
+ * is inserted just after the ACTUAL_PIXELS frame:
+ *    newActualPixelsToPixels = pixelTransform -> oldActualPixelsToPixels
+ * This is appropriate for shitting a WCS, e.g. when writing FITS metadata for a subimage.
+ *
+ * If modifyActualPixels is false or the ACTUAL_PIXELS frame does not exist then pixelTransform
+ * is inserted just after the PIXELS frame:
+ *    newPixelsToIwc = pixelTransform -> oldPixelsToIwc
+ * This is appropriate for inserting a model for optical distortion.
+ *
+ * Other than the change described above, the new SkyWcs will be just like the old one.
+ *
+ * @param[in] pixelTransform  Transform to insert
+ * param[in] modifyActualPixels  Location at which to insert the transform;
+ *    if true and the ACTUAL_PIXELS frame is present then insert just after the ACTUAL_PIXELS frame,
+ *    else insert just after the PIXELS frame.
+ */
+std::shared_ptr<SkyWcs> makeModifiedWcs(TransformPoint2ToPoint2 const &pixelTransform, SkyWcs const &wcs,
+                                        bool modifyActualPixels);
+
+/**
+ *  Construct a TAN-SIP WCS
+ *
+ * Construct a TAN WCS with forward SIP distortion terms and an iterative inverse.
+ *
+ * @param[in] crpix    Center of projection on the CCD using the LSST convention:
+ *                      0, 0 is the lower left pixel of the parent image
+ * @param[in] crval    Center of projection on the sky
+ * @param[in] cdMatrix CD matrix, where element (i-1, j-1) corresponds to FITS keyword CDi_j
+ *                      and i, j have range [1, 2]. May be computed by calling makeCdMatrix.
+ * @param[in] sipA     Forward distortion matrix for axis 1
+ * @param[in] sipB     Forward distortion matrix for axis 2
+ */
+std::shared_ptr<SkyWcs> makeTanSipWcs(Point2D const &crpix, coord::IcrsCoord const &crval,
+                                      Eigen::Matrix2d const &cdMatrix, Eigen::MatrixXd const &sipA,
+                                      Eigen::MatrixXd const &sipB);
+
+/**
+ *  Construct a TAN-SIP WCS
+ *
+ * Construct a TAN WCS with SIP distortion terms. Eventually this may pay attention to the
+ * inverse SIP matrices, but for now they are ignored and the inverse is strictly iterative.
+ *
+ * @param[in] crpix    Center of projection on the CCD using the LSST convention:
+ *                      0, 0 is the lower left pixel of the parent image
+ * @param[in] crval    Center of projection on the sky
+ * @param[in] cdMatrix CD matrix, where element (i-1, j-1) corresponds to FITS keyword CDi_j
+ *                      and i, j have range [1, 2]. May be computed by calling makeCdMatrix.
+ * @param[in] sipA     Forward distortion matrix for axis 1
+ * @param[in] sipB     Forward distortion matrix for axis 2
+ * @param[in] sipAp    Reverse distortion matrix for axis 1 (ignored)
+ * @param[in] sipBp    Reverse distortion matrix for axis 2 (ignored)
+ */
```

Comment:
Thanks. They are no longer ignored and I missed that bit when cleaning up the docs.

For the record: initially they were ignored, but I wanted them in case they were eventually used (either as stated, or as hints to a solver). The iterative solver proved too fragile so AST was updated so they are always used.